### PR TITLE
Introduce CONTACTDB helpers

### LINKED ISSUE #311: Create helper methods for the /contactdb endpoint
#### Issue Summary

Please see `USE_CASES.md` for current example of existing helper methods.

You may find simple example code for all of the `/contactdb` endpoint functionality in the `USAGE.md` file under `CONTACTDB` in the Table of Contents. Complete interactive documentation can be found [here](https://sendgrid.com/docs/API_Reference/api_v3.html).

Before writing any code, we suggest you review our `CONTRIBUTING.md` file and propose your solution for discussion here in the comments. Thank you!

#### Aceptance Criteria

* helper methods are created for all documented /contactdb functionality
* tests pass locally, in Travis CI and using Docker
* updated `USE_CASES.md` or `/use-cases` documentation

### PR DESCRIPTION
**Pull request by DearGrenadier**
# Fixes #311 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Add Segment helper
- Add SegmentCondition helper
- Update USAGE examples
- Add proper test

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=352) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=352) before we can accept your contribution.<br/><hr/>**Dima Nazarchuk** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=352) it.</sub>

**Comment by DearGrenadier** [comment]
Hello @thinkingserious, just wondering, could we merge this PR? I think that already implemented helpers are sufficient for CONTACTDB endpoints, remaining cases are pretty straightforward and, how I feel, don't require any additional syntax sugar.

File: lib/sendgrid/helpers/contactdb/base_helper.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module SendGrid
+  class BaseHelper
+    private
+
+    def sanitize(**data)
+      data.delete_if { |_, value| value.to_s.strip == '' }
```
Comment (thinkingserious) [inline lib/sendgrid/helpers/contactdb/base_helper.rb:6]: How about `nil` values?

**Comment by childish-sambino** [comment]
Closing due to inactivity. Please ping here if work is to continue.

**Pull request closed without merging**